File: python/package/converter.py
import sys


class RomanNumeralConverter:
    MIN_ROMAN_NUMERAL = int(1)
    MAX_ROMAN_NUMERAL = int(3999)

    input_number = 0

    def CheckInputNumber(self):
        if (self.MIN_ROMAN_NUMERAL > self.input_number or self.input_number > self.MAX_ROMAN_NUMERAL):
            print('{} is not proper input.'.format(self.input_number))
            self.AddArabianNumber()

        self.ApplyRomanConverter()

    def AddArabianNumber(self):
        print(
            'Please enter the integer again in the range of 1 to 3999 (Enter 0 to exit).')
        self.input_number = input('Please enter a value -> ')
        if self.input_number:
            self.CheckInputNumber()
        else:
            sys.exit()

    def ApplyRomanConverter(self):
        self.i = 0
        self.romanNumber = ''
        self.correspondingValue = (
            (100, 'M'), (900, 'CM'), (500, 'D'), (400, 'CD'), (100, 'C'), (90, 'XC'),
            (50, 'L'), (40, 'XL'), (10, 'X'), (9, 'IX'), (5, 'V'), (4, 'IV'), (1, 'I')
        )
        while self.input_number:
            while (self.input_number / self.correspondingValue[self.i][0]):
                self.romanNumber += self.correspondingValue[self.i][1]
                self.input_number -= self.correspondingValue[self.i][0]

            self.i += 1

        print(self.romanNumber)

File: cpp/converter.cpp
#include <iostream>
#include <string>
#include <limits>
#include <stdlib.h>
#include <vector>
#include "converter.hpp"

void RomanNumeralConverter::CheckInputNumber()
{
    if (MIN_ROMAN_NUMERAL > this->m_inputNumber || this->m_inputNumber > MAX_ROMAN_NUMERAL)
    {
        std::cout << this->m_inputNumber << " is not proper input." << std::endl;
        this->AddArabinNumber();
    }

    this->ApplyRomanConverter();
}

void RomanNumeralConverter::AddArabinNumber()
{
    std::cout << "Please enter the integer again in the range of 1 to 3999 (Enter 0 to exit)." << std::endl;
    std::cout << "Please enter a value -> ";
    std::cin >> this->m_inputNumber;

    if (m_inputNumber)
    {
        this->CheckInputNumber();
    }
    else
    {
        std::exit(EXIT_FAILURE);
    }
}

void RomanNumeralConverter::ApplyRomanConverter()
{
    int i = 0;
    std::string romanNumber;
    std::vector<std::pair<int, std::string>> correspondingValue =
    {
        {1000, "M"}, {900, "CM"}, {500, "D"}, {400, "CD"}, {100, "C"}, {90, "XC"},
        {50, "L"}, {40, "XL"}, {10, "X"}, {9 , "IX"}, {5, "V"}, {4, "IV"}, {1, "I"}
    };

    while (this->m_inputNumber)
    {
        while (m_inputNumber / correspondingValue[i].first)
        {
            romanNumber += correspondingValue[i].second;
            this->m_inputNumber -= correspondingValue[i].first;
        }
        i++;
    }

    std::cout << romanNumber;
}

